icons/src/fullcolor/render-bitmaps-verbose.py

```diff
@@ -238,60 +221,278 @@ def characters(self, chars):
 
     # If invocation does not include a file name, process all SVGs in listed sources
     else:
-        print("Rendering from path \"" + SRC + "\"")
-        if not os.path.exists(OUTPUT_DIRECTORY):
-            os.mkdir(OUTPUT_DIRECTORY)
+        print("Rendering from path \"" + category_to_render + "\"")
+        if not os.path.exists(args.output_path):
+            os.mkdir(args.output_path)
 
-        for file in os.listdir(SRC):
+        # this is the loop for the files in the given directory
+        for file in os.listdir(category_to_render):
             if file[-4:] == '.svg':
-                file = os.path.join(SRC, file)
-                handler = ContentHandler(file)
+                file = os.path.join(category_to_render, file)
+                #
+                # ContentHandler() implements an interface for xml.sax.parse()
+                #
+                handler = ContentHandler(file, args)
+                #
+                # xml.sax.parse() uses the various ContentHandler() methods behind the scenes
+                #
                 xml.sax.parse(open(file), handler)
         
         return True
 
-    
 
-parser = argparse.ArgumentParser(description='Render icons from SVG to PNG')
+def check_dependencies(args):
+
+    # assert ('linux' in sys.platform), "This code runs on Linux only."
+    assert (os.path.isfile(args.inkscape_path)), "Expected to find Inkscape at " + args.inkscape_path + ", but file does not exist."
+    assert (os.path.isfile(args.optipng_path)), "Expected to find OptiPNG at " + args.optipng_path + " but file does not exist."
+    assert (os.path.isdir(args.output_path)), "Expected to find output directory at " + args.output_path + ", but directory does not exist."
```

Comment:
At line 226 `args.output_path` is created if it does not exist. Is this assert correct?

Reply:
That’s probably a holdover from something else. (The assertions are from when I was initially debugging the version of the script over at `Mint-y-icons`.)